### ISSUE TITLE
Blog benchmark crash on OSX

### ISSUE THREAD

**Issue by djones6** [IBM-Swift/Kitura#786]
Kitura 1.0 and Kitura-net 1.0.1 crashes when running the Blog benchmark on OSX after a variable amount of time:
[Output](https://gist.github.com/djones6/5b0c1053f8990a9ea0a4f3d3b44c62d5)
So far I have been able to readily recreate this crash (with this stack trace) on OSX. It usually fails within a minute.
On Linux with the GCD keepalive implementation I have been able to recreate the same crash, however it took much longer (over 10 minutes).  I cannot recreate with epoll.

Recreate steps

1. Build driver
`git clone https://github.com/wg/wrk`
`cd wrk`
`make`

2. Build benchmark
`git clone https://github.com/djones6/Server-Side-Swift-Benchmarking`
`cd Server-Side-Swift-Benchmarking/KituraPress`
`swift build`

3. terminal 1:
`ulimit -c unlimited`
`.build/debug/FairBlog`

4. terminal 2:
`wrk -c20 -t4 -d10s http://localhost:8080/blog`
(repeat until server crashes)

**Comment by djones6** [comment]
The failure occurs in `IncomingSocketHandler.socketReaderQueue(fd : Int32)`
```
5    libswiftCore.dylib                 0x000000010fc44970 specialized _fatalErrorMessage(StaticString, StaticString, StaticString, UInt, flags : UInt32) -> Never + 96
6    libswiftCore.dylib                 0x000000010faf9830 _ArrayBuffer._checkInoutAndNativeTypeCheckedBounds(Int, wasNativeTypeChecked : Bool) -> () + 300
7    libswiftCore.dylib                 0x000000010fb0a510 Array.subscript.getter + 103
8    FairBlog                           0x000000010f882bf0 IncomingSocketHandler.socketReaderQueue(fd : Int32) -> DispatchQueue + 159
9    FairBlog                           0x000000010f883e00 IncomingSocketHandler.handleBufferedReadData() -> () + 86
10   FairBlog                           0x000000010f8a9d50 IncomingHTTPSocketProcessor.keepAlive() -> () + 260
11   FairBlog                           0x000000010f8a6160 HTTPServerResponse.end() throws -> () + 164
```
It occurs when the file descriptor we pass in to this function is negative (-1) which probably indicates some sort of failure we didn't expect.
```
Getting queue: 9 -> 1
Getting queue: 16 -> 0
Getting queue: 24 -> 0
...
Getting queue: -1 -> -1
fatal error: Index out of range
```

**Comment by djones6** [comment]
The Linux case is the same - backtrace in the core shows the same situation:
```
    frame #2: 0x00007fc246d4db12 libswiftCore.so`Swift.Array.subscript.getter : (Swift.Int) -> A + 34
    frame #3: 0x00000000004727db FairBlog`IncomingSocketHandler.socketReaderQueue(fd=-1, self=<unavailable>) -> DispatchQueue + 155 at IncomingSocketHandler.swift:49
    frame #4: 0x0000000000473f2b FairBlog`IncomingSocketHandler.handleBufferedReadData(self=<unavailable>) -> () + 75 at IncomingSocketHandler.swift:149
```

**Comment by djones6** [comment]
Seems this may have just been fixed by #784 
I'll try the patch from that branch and confirm whether it fixes this problem.

**Comment by djones6** [comment]
Fixed by 784.
